front/src/common/Map/WarpedMap/SimulationWarpedMap.tsx

```diff
@@ -242,26 +254,36 @@ const SimulationWarpedMap = ({ collapsed }: { collapsed?: boolean }) => {
     state,
   ]);
 
+  const updateWarpedMapState = (coordinates: GeoJsonLineStringValue) => {
+    const path = lineString(coordinates);
+    const pathBBox = bbox(path) as BBox2d;
+    const { warpedBBox, transform } = getWarping(path);
+    setState({ type: 'pathLoaded', path, pathBBox, warpedBBox, transform });
+  };
+
   /**
    * This effect handles loading the simulation path, and retrieve the warping function:
    */
   useEffect(() => {
     setState({ type: 'loading' });
-    getPath({ pathfindingId: pathfindingID })
-      .then(({ data, isError, error }) => {
-        if (isError) {
-          setState({ type: 'error', message: error as string });
-        } else if (!data?.geographic?.coordinates) {
-          setState({ type: 'error', message: 'No coordinates' });
-        } else {
-          const path = lineString(data?.geographic?.coordinates as Position[]);
-          const pathBBox = bbox(path) as BBox2d;
-          const { warpedBBox, transform } = getWarping(path);
-
-          setState({ type: 'pathLoaded', path, pathBBox, warpedBBox, transform });
-        }
-      })
-      .catch((error) => setState({ type: 'error', message: error }));
+
+    // V2
+    if (pathProperties) {
```

Comment:
If the core functionality depends on whether Train Schedule V2 is activated, it would be more reliable to use the `getTrainScheduleV2Activated` selector like so: `const trainScheduleV2Activated = useSelector(getTrainScheduleV2Activated);`. 
Relying on the pathProperties prop could potentially mask bugs, such as when it is undefined due to unexpected reasons. In such cases, even if Train Schedule V2 is activated, the V1 code block would be executed, which is not the expected behavior.